backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/EdgeComponent.java

```diff
@@ -107,22 +107,28 @@ public Element render() {
                                     .map(Edge::getType)
                                     .orElse("edge:straight"); //$NON-NLS-1$
 
-                            List<Position> routingPoints = optionalPreviousEdge
-                                    .map(Edge::getRoutingPoints)
-                                    .orElseGet(()-> edgeRoutingPointsProvider.getRoutingPoints(sourceNode, targetNode));
+                            List<Position> routingPoints;
```

Comment:
Not directly related to this change, but the `render()` method is getting out of hands, with almost 100 lines, 3 nested for loops with up to 3 additional `if`s...

Reply:
I agree with you. This class will be refactored by an other PR that will come quickly.